### PR TITLE
Refactor Union Pack and UnPack methods

### PR DESCRIPTION
**Pull request by tigrato**
Refactor Union Pack and UnPack methods
Remove append usage to increase code efficiency when dealing with large vectors

### CODE REVIEW

File: src/idl_gen_go.cpp [hunk 1]
```diff
@@ -43,7 +43,7 @@ static std::string GeneratedFileName(const std::string &path,
 namespace go {
 
 // see https://golang.org/ref/spec#Keywords
-static const char * const g_golang_keywords[] = {
+static const char *const g_golang_keywords[] = {
```
Comment (iceb0y) [inline src/idl_gen_go.cpp]: Formatting irrelevant code can create unnecessary conflicts if the parent PR is pending for a long time. I would suggest to keep them as-is in this PR. You can send separate PR for reformatting.

File: src/idl_gen_go.cpp [hunk 2]
```diff
@@ -968,14 +1005,15 @@ class GoGenerator : public BaseGenerator {
         code += "\tt." + field_name_camel + " = rcv." + field_name_camel +
                 "Bytes()\n";
       } else if (field.value.type.base_type == BASE_TYPE_VECTOR) {
-        code += "\tfor j := 0; j < rcv." + field_name_camel +
-                "Length(); j++ {\n";
+        code += "\t" + length + " := rcv." + field_name_camel + "Length()\n";
+        code += "\tt." + field_name_camel + " = make(" +
+                NativeType(field.value.type) + ",  " + length + ")\n";
```
Comment (iceb0y) [inline src/idl_gen_go.cpp]: redundant space

File: src/idl_gen_go.cpp [hunk 3]
```diff
@@ -986,29 +1024,20 @@ class GoGenerator : public BaseGenerator {
...
-          code += "\t\tx := " + ev.union_type.struct_def->name + "{}\n";
-          code += "\t\tif rcv." + MakeCamel(field.name) + "(&x._tab) {\n";
-          code += "\t\t\tt." + MakeCamel(field.name) + " = &" +
-                  WrapInNameSpaceAndTrack(
-                      enum_def.defined_namespace, NativeName(enum_def)) +
-                  "{ Type: " + enum_def.name + ev.name +
-                  ", Value: x.UnPack() }\n";
-          code += "\t\t}\n";
-        }
+        std::string field_table = "tb" + MakeCamel(field.name);
```
Comment (iceb0y) [inline src/idl_gen_go.cpp]: I would suggest to use suffix instead of prefix to be consistent with the rest of the code. i.e.

`MakeCamel(field.name, false) + "Table"`

File: src/idl_gen_go.cpp [hunk 4]
```diff
@@ -986,29 +1024,20 @@ class GoGenerator : public BaseGenerator {
...
+        code += "\t" + field_table + " := flatbuffers.Table{}\n";
+        code +=
+            "\tif rcv." + MakeCamel(field.name) + "(&" + field_table + "){\n";
```
Comment (iceb0y) [inline src/idl_gen_go.cpp]: Missing space between `)` and `{`.

**Comment by tigrato** [comment]
Thanks @iceb0y for the CR.

I can't merge this PR.
